src/openms/include/OpenMS/ANALYSIS/ID/PeptideIndexing.h

```diff
@@ -677,8 +699,155 @@ namespace OpenMS
+         proteins.cacheChunk(PROTEIN_CACHE_SIZE);
+         bool has_active_data = proteins.activateCache();
+
+         if (!has_active_data) break;
+         auto prot_count = (SignedSize)proteins.chunkSize();
+
+         {
+           for (SignedSize i = 0; i < prot_count; ++i)
+           {
+             String seq = proteins.chunkAt(i).identifier;
+             String seq_lower = seq;
+             seq_lower.toLower();
+
+             // remove all non-alphanumeric characters
+             seq_lower.erase(std::remove_if(seq_lower.begin(), seq_lower.end(),
+                                            [](char c) { return !std::isalpha(c); } ), seq_lower.end());
+
+             // search for case insensitive occurrence of the possible decoys
+             for (auto pair : decoy_count)
+             {
+               if (seq_lower.hasPrefix(pair.first))
+               {
+                 // count observed (ignoring case)
+                 decoy_count[pair.first].first++;
+                 // store observed (case sensitive)
+                 std::string seq_decoy = StringUtils::prefix(seq, pair.first.length());
+                 decoy_case_sensitive[pair.first] = seq_decoy;
+               }
+               else if (seq_lower.hasSuffix(pair.first))
+               {
+                 // count observed (ignoring case)
+                 decoy_count[pair.first].second++;
+                 // store observed (case sensitive)
+                 std::string seq_decoy = StringUtils::suffix(seq, pair.first.length());
+                 decoy_case_sensitive[pair.first] = seq_decoy;
+               }
+             }
+           }
+         }
+       }
+
+       // DEBUG only: print counts of found decoys
+       for (auto &a : decoy_count) LOG_DEBUG << a.first << "\t" << a.second.first << "\t" << a.second.second << std::endl;
+
+       return determineDecoyStringFromCounts_(decoy_count, decoy_case_sensitive, decoy_string, is_prefix);
+     }
+
+     ExitCodes determineDecoyStringFromCounts_(const MapDecoyStringToPrefixSuffixCount & decoy_count,
+                                     const MapCaseInsensitiveToCaseSensitiveDecoyString & decoy_case_sensitive,
+                                     std::string & decoy_string,
+                                     bool & is_prefix) const
+     {
+       int all_prefix_occur(0), all_suffix_occur(0);
+
+       // count observed decoy prefixes
+       for (const auto &pair : decoy_count)
+       {
+         all_prefix_occur += pair.second.first;
+       }
+
+       // count observed decoy suffixes
+       for (const auto &pair : decoy_count)
+       {
+         all_suffix_occur += pair.second.second;
+       }
+
+       if (all_prefix_occur == all_suffix_occur)
+       {
+         LOG_ERROR << "Unable to determine decoy string" << std::endl;
+         return DECOYSTRING_EMPTY;
+       }
+
+       // if a decoy prefix occured at least 80% -> set it as prefix decoy
+       for (const auto & pair : decoy_count)
```

Comment:
formatting: please attach the `&` to the type (no spaces), ie. `const auto& pair`

occurs in a few places

Reply:
I was told that there's a friendly war about this in OpenMS ;)
Will do.